HotelReservation/HotelReservationTests/ReservationSteps/Mail/SendingMailProcessTests.cs

```diff
@@ -14,31 +15,56 @@ class SendingMailProcessTests
         SendingMailProcess _subject;
         ConsolePrinter _consolePrinterDouble;
         StepInput _name;
-        StepInput _email;
+        StepInput _mail;
 
         [SetUp]
         public void BeforeTest()
         {
             _consolePrinterDouble = A.Fake<ConsolePrinter>();
             _subject = new SendingMailProcess(_consolePrinterDouble);
             _name = new StepInput(InputType.Name);
-            _email = new StepInput(InputType.EmailAddress);
+            _mail = new StepInput(InputType.EmailAddress);
         }
 
         [Test]
         public void ShouldProvideSendingMailProcessInputs()
         {
-            var stepInputs = new List<StepInput> { _name, _email };
+            var stepInputs = new List<StepInput> { _name, _mail };
 
-            _subject.GetStepInputs().Should().BeEquivalentTo(stepInputs);
+            _subject.GetRequiredStepInputs().Should().BeEquivalentTo(stepInputs);
         }
 
         [Test]
         public void ExecuteShouldCallConsolePrinterWithStepName()
         {
-            _subject.Execute(new List<StepInput>());
+            _subject.Execute(new List<StepInput> { _name, _mail });
 
             A.CallTo(() => _consolePrinterDouble.Write("----==== SENDING MAIL PROCESS ====----")).MustHaveHappened();
         }
+
+        [Test]
+        public void ShouldCallConsolePrinterWithProvidedArgument()
+        {
+            const string nameValue = "Test name value";
+            const string mailValue = "Test mail value";
+
+            _name.Value = nameValue;
+            _mail.Value = mailValue;
+
+            _subject.Execute(new List<StepInput> { _name, _mail });
+
+            A.CallTo(() => _consolePrinterDouble.Write(nameValue)).MustHaveHappened();
```

Comment:
To powinny być dwa testy. Kiedy ten test nie przejdzie nie wiesz od razu, którego asserta to dotyczy.

Reply:
Fixed